### PR TITLE
Add reprocess_geocode_failures management command

### PR DESCRIPTION
**Pull request by jwalgran**
## Overview

A bug in the geocoding process left us with a significant number of items that were not geocoded and are now in the ERROR_MATCHING status. This management changes the status on these items and then re-runs the geocode and match processing on the lists to which they belong.

Connects #403 
Connects #428 

## Notes

Dedupe raises an exception if you attempt to match an empty list. This PR includes a commit that avoids the error by checking that there is at least on item in one of the geocoded statuses before attempting to match and setting a `no_geocoded_items` flag accordingly.

We ran into CI build failures on this branch that were resolved by defining `COMPOSE_PROJECT_NAME` in the Jenkinsfile.

### Setup

* Apply this diff to generate some no result geocodes when processing the fixture data
```diff
diff --git a/src/django/api/processing.py b/src/django/api/processing.py
index 0656884..01e55a1 100644
--- a/src/django/api/processing.py
+++ b/src/django/api/processing.py
@@ -129,8 +129,13 @@ def geocode_facility_list_item(item):
                 'finished_at': str(datetime.utcnow()),
                })
         else:
-            item.status = FacilityListItem.GEOCODED
-            item.geocoded_address = item.address
+            if item.id % 10 == 0:
+                item.status = FacilityListItem.GEOCODED_NO_RESULTS
+                item.geocoded_address = None
+                item.geocoded_point = None
+            else:
+                item.status = FacilityListItem.GEOCODED
+                item.geocoded_address = item.address
             item.processing_results.append({
                 'action': ProcessingAction.GEOCODE,
                 'started_at': started,


```
* Run `./scripts/manage resetdb` and `./scripts/manage processfixtures`

### Test

* Run `./scripts/manage reprocess_geocode_failures`. Run the processing commands that are printed to the console.
```
./scripts/manage batch_process --list-id 2 --action geocode && \
./scripts/manage batch_process --list-id 2 --action match && \
./scripts/manage batch_process --list-id 3 --action geocode && \
./scripts/manage batch_process --list-id 3 --action match && \
./scripts/manage batch_process --list-id 4 --action geocode && \
./scripts/manage batch_process --list-id 4 --action match && \
./scripts/manage batch_process --list-id 5 --action geocode && \
./scripts/manage batch_process --list-id 5 --action match && \
./scripts/manage batch_process --list-id 6 --action geocode && \
./scripts/manage batch_process --list-id 6 --action match && \
./scripts/manage batch_process --list-id 7 --action geocode && \
./scripts/manage batch_process --list-id 7 --action match && \
./scripts/manage batch_process --list-id 8 --action geocode && \
./scripts/manage batch_process --list-id 8 --action match && \
./scripts/manage batch_process --list-id 9 --action geocode && \
./scripts/manage batch_process --list-id 9 --action match && \
./scripts/manage batch_process --list-id 10 --action geocode && \
./scripts/manage batch_process --list-id 10 --action match && \
./scripts/manage batch_process --list-id 11 --action geocode && \
./scripts/manage batch_process --list-id 11 --action match && \
./scripts/manage batch_process --list-id 12 --action geocode && \
./scripts/manage batch_process --list-id 12 --action match && \
./scripts/manage batch_process --list-id 13 --action geocode && \
./scripts/manage batch_process --list-id 13 --action match && \
./scripts/manage batch_process --list-id 14 --action geocode && \
./scripts/manage batch_process --list-id 14 --action match && \
./scripts/manage batch_process --list-id 15 --action geocode && \
./scripts/manage batch_process --list-id 15 --action match
```

* Verify that the process completes without error.

## Checklist

- [x] `fixup!` commits have been squashed
- [x] CI passes after rebase
- [x] CHANGELOG.md updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines

### CODE REVIEW

**Comment by kellyi** [comment]
Taking a look at this.

**Comment by kellyi** [comment]
Went through the testing instructions and here is the output:

```
vagrant@vagrant:/vagrant$ ./scripts/manage reprocess_geocode_failures                                                                                                                                                                  
Starting vagrant_database_1 ... done                                                                                                                                                                                                   
Status counts:                                                                                                                                                                                                                         
  ERROR_MATCHING 91                                                                                                                                                                                                                    
  MATCHED 839
  POTENTIAL_MATCH 2
Geocoding 91 items
GEOCODED_NO_RESULTS 570 CN NO.18,JINSHAN ROAD,BALI TOWN,ECONOMIC & DEVELOPING AREA.,YANGZHOU,JIANGSU
GEOCODED_NO_RESULTS 620 TW 40-19, Bo Kon Kun, Shi-Fu Village, Shan Yih Hsiang, Miaoli Hsien, 36744, Taiwan
...
INFO:dedupe.api:0 records
Status counts:
  ERROR_MATCHING 2
  MATCHED 925
  POTENTIAL_MATCH 5
vagrant@vagrant:/vagrant$ 
```

File: src/django/api/management/commands/reprocess_geocode_failures.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                            geocode_facility_list_item,
+                            match_facility_list_items,
+                            save_match_details)
+
+LINE_ITEM_ACTIONS = {
+    ProcessingAction.PARSE: parse_facility_list_item,
+    ProcessingAction.GEOCODE: geocode_facility_list_item,
+}
+
+LIST_ACTIONS = set([ProcessingAction.MATCH])
```
Comment (kellyi) [inline src/django/api/management/commands/reprocess_geocode_failures.py]: Could this just be `{ProcessingAction.MATCH}`?

Comment (jwalgran) [reply src/django/api/management/commands/reprocess_geocode_failures.py]: Oops. These constants were copy pasted from a previous management command and are not even used. Will remove.

**Comment by jwalgran** [comment]
I pushed a significant change to this management command.

The original version of this management command ran all the processing in the context of the command itself. This worked correctly for test data, but the process was killed when trying to run it on production-sized data.

The batch processing tasks are designed to only work on items in a specific status so we can safely change the status on just the items we need to process and then resubmit the entire list to the geocoding and match tasks.

We have added a `skip_parse` option to `submit_jobs` as a convenience since we know that we will not have to re-parse any items.

This is ready for a re-review.

**Comment by jwalgran** [comment]
>The only difference is that the testing instructions indicate chaining the geocode and match commands using &&

That was by design. I added the `&&` in the testing instructions to make things easier on the tester.

I ran this command on a staging deploy with realistic data and saw jobs being correctly scheduled in AWS Batch.

Thanks for the double review.

**Pull request merged**